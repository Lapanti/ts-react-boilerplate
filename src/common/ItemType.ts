enum ItemType {
  STORY = 'story',
  JOB = 'job',
  COMMENT = 'comment',
  POLL = 'poll',
  POLLOPT = 'pollopt',
}

export default ItemType;
