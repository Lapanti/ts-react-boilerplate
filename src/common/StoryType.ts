enum StoryType {
  NEW = 'new',
  TOP = 'top',
  BEST = 'best',
}

export default StoryType;
