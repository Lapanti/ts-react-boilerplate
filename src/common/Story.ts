import ItemType from './ItemType';

export default interface Story {
  readonly by: string;
  readonly descendants: number;
  readonly id: number;
  readonly kids: number[];
  readonly score: number;
  readonly time: number;
  readonly title: string;
  readonly type: ItemType.STORY;
  readonly url: string;
}
