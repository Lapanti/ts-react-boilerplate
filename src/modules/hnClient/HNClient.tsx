import * as React from 'react';
import { RouteComponentProps } from 'react-router-dom';
import { connect } from 'react-redux';

import styled, { withProps } from '../../theme/styled';
import { State } from '../../redux/reducer';
import StoryType from '../../common/StoryType';
import Story from '../../common/Story';
import Loader from '../../components/Loader';
import Header from './header/Header';

const ClientContainer = withProps<{ rows: number }>(styled.main)`
  display: grid;
  font-family: 'Roboto', arial;
  grid-template-rows: 40px repeat(${props => props.rows}, 60px);
  width: 100%;
`;

export interface HNClientState {
  readonly storyType: StoryType;
  readonly stories: Story[];
  readonly loading: boolean;
}

export interface HNClientDispatch {}

export type HNClientProps = HNClientState &
  HNClientDispatch &
  RouteComponentProps<never>;

export const HNClient: React.StatelessComponent<HNClientProps> = ({
  storyType,
  stories,
  loading,
}) => (
  <ClientContainer rows={stories.length}>
    {loading && <Loader />}
    <Header />
    {stories.map(story => <p key={story.id}>{story.title}</p>)}
  </ClientContainer>
);

export default connect<HNClientState, HNClientDispatch, {}, State>(
  state => ({
    storyType: state.hnClient.storyType,
    stories: state.hnClient.stories,
    loading: state.hnClient.loading,
  }),
  {},
)(HNClient);
