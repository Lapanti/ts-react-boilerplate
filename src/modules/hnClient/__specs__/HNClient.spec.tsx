import * as React from 'react';
import { shallow } from 'enzyme';
import createHistory from 'history/createBrowserHistory';

import { HNClient } from '../HNClient';
import Story from '../../../common/Story';
import StoryType from '../../../common/StoryType';
import ItemType from '../../../common/ItemType';

describe('<HNClient />', () => {
  const testStory1: Story = {
    by: 'me',
    descendants: 0,
    id: 123456,
    kids: [],
    score: 0,
    time: 1,
    title: 'Test',
    type: ItemType.STORY,
    url: '',
  };
  const testStory2: Story = {
    by: 'me',
    descendants: 0,
    id: 234567,
    kids: [],
    score: 1,
    time: 1,
    title: 'Test 2',
    type: ItemType.STORY,
    url: '',
  };
  const wrapperMinimalProps = shallow(
    <HNClient
      storyType={StoryType.NEW}
      stories={[]}
      loading={false}
      match={{ params: undefined, isExact: true, path: '', url: '' }}
      location={{ pathname: '', search: '', state: {}, hash: '', key: '' }}
      history={createHistory()}
    />,
  );
  const wrapperMaximumProps = shallow(
    <HNClient
      storyType={StoryType.NEW}
      stories={[testStory1, testStory2]}
      loading={false}
      match={{ params: undefined, isExact: true, path: '', url: '' }}
      location={{ pathname: '', search: '', state: {}, hash: '', key: '' }}
      history={createHistory()}
    />,
  );

  it('should render with correct props', () => {
    expect(wrapperMinimalProps).toMatchSnapshot();
    expect(wrapperMaximumProps).toMatchSnapshot();
  });
});
