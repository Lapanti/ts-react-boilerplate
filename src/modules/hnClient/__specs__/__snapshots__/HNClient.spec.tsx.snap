// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HNClient /> should render with correct props 1`] = `
<styled.main
  rows={0}
>
  <Header />
</styled.main>
`;

exports[`<HNClient /> should render with correct props 2`] = `
<styled.main
  rows={2}
>
  <Header />
  <StoryItem
    index={0}
    key="123456"
    story={
      Object {
        "by": "me",
        "descendants": 0,
        "id": 123456,
        "kids": Array [],
        "score": 0,
        "time": 1,
        "title": "Test",
        "type": "story",
        "url": "",
      }
    }
  />
  <StoryItem
    index={1}
    key="234567"
    story={
      Object {
        "by": "me",
        "descendants": 0,
        "id": 234567,
        "kids": Array [],
        "score": 1,
        "time": 1,
        "title": "Test 2",
        "type": "story",
        "url": "",
      }
    }
  />
</styled.main>
`;
